frontend/test/views/AccountView.spec.js

```diff
@@ -83,48 +88,105 @@ describe('AccountView.vue', () => {
 
-    sectionBox = wrapper.findComponent('[data-test=credentials]');
-    expect(sectionBox.props('content')[1].value).toEqual([wrapper.vm.user.clientSecret]);
-    await sectionBox.trigger('click');
+    credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal(wrapper.vm.clientSecret);
+    await credentialsBox.vm.$emit('display-secret');
     await wrapper.vm.$nextTick();
 
-    sectionBox = wrapper.findComponent('[data-test=credentials]');
-    expect(sectionBox.props('content')[1].value).toEqual([wrapper.vm.user.clientSecret]);
+    credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal(wrapper.vm.clientSecret);
   });
 
+  it('should use clientSecret prop in computed secretValue', async () => {
+    const wrapper = getMountedComponent();
 
-  it('should return ["<empty>"] when user.client_secret is not set', () => {
-    sandbox.restore();
-    sandbox = sinon.createSandbox();
-    mockUser = {
-      clientSecret: '',
-    };
+    let credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal('<click to show>');
+
+    getAPICredentialsStub.returns({client_secret: 'updatedSecret'});
+    await credentialsBox.vm.$emit('generate-secret');
+    await wrapper.vm.$nextTick();
 
-    const getUserEmptyStub = sandbox.stub(authStore, 'getUser');
-    getUserEmptyStub.returns(mockUser);
+    credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal('<click to show>');
 
+    await credentialsBox.vm.$emit('display-secret');
+    await wrapper.vm.$nextTick();
+
+    credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal('updatedSecret');
+  });
+
+  it('should update showSecretValue to true when updateSecretValue is called', () => {
     const wrapper = getMountedComponent();
+    wrapper.vm.updateSecretValue();
+    expect(wrapper.vm.showSecretValue).to.be.true;
+  });
+
+  it('should update clientSecret on multiple generateSecret calls', async () => {
+    getAPICredentialsStub.onFirstCall().returns({client_secret: 'newSecret1'});
+    getAPICredentialsStub.onSecondCall().returns({client_secret: 'newSecret2'});
 
-    let sectionBox = wrapper.findComponent('[data-test=credentials]');
+    const wrapper = getMountedComponent();
+    const credentialsBox = wrapper.findComponent('[data-test=credentials]');
 
-    sectionBox = wrapper.findComponent('[data-test=credentials]');
-    expect(sectionBox.props('content')[1].value).toEqual(['<empty>']);
+    await credentialsBox.vm.$emit('generateSecret');
+    await wrapper.vm.$nextTick();
+    expect(generateSecretStub.calledOnce).to.be.true;
+    expect(getAPICredentialsStub.calledOnce).to.be.true;
+    expect(wrapper.vm.clientSecret).to.equal('newSecret1');
```

Comment:
https://github.com/uab-cgds-worthey/rosalution/pull/58/files/0fff0226776a4f55ded1ff64a25f4f6260d1ba6a#r1140234713 As mentioned from this previous pull request review;  verifying the VueJS Component data property is an implementation detail that should not be tested.  https://test-utils.vuejs.org/guide/essentials/easy-to-test.html#do-not-test-implementation-details 

In this case, 'clientSecret' VueJS Component data is utilized in a VueJS Component computed value, 'secretValue', which then is provided as a prop to the 'CredentialsBox' VueJS component you have created.  Referring to the article shared, what is the observable non-implementation detail that can be checked to verify this?